Src/AutoFixture.xUnit.net.UnitTest/Scenario.cs

```diff
@@ -141,5 +141,85 @@ public void FavorArraysCausesArrayConstructorToBeInjectedWithFrozenItems([Frozen
         {
             Assert.Same(p1, p2);
         }
+
+        [Theory, AutoData]
+        public void FreezeFirstParameterShouldAssignSameInstanceToSecondParameter(
+            [Freeze]string p1,
+            string p2)
+        {
+            Assert.Same(p1, p2);
+        }
+
+        [Theory, AutoData]
+        public void FreezeFirstParameterByExactTypeShouldAssignSameInstanceToSecondParameter(
+            [Freeze(By = Matching.ExactType)]ConcreteType p1,
+            ConcreteType p2)
+        {
+            Assert.Same(p1, p2);
+        }
+
+        [Theory, AutoData]
+        public void FreezeFirstParameterByDirectBaseTypeShouldAssignSameInstanceToSecondParameter(
+            [Freeze(By = Matching.DirectBaseType)]ConcreteType p1,
+            AbstractType p2)
+        {
+            Assert.Same(p1, p2);
+        }
+
+        [Theory, AutoData]
+        public void FreezeFirstParameterByInterfaceShouldAssignSameInstanceToSecondParameter(
+            [Freeze(By = Matching.ImplementedInterfaces)]NoopInterfaceImplementer p1,
+            IInterface p2)
+        {
+            Assert.Same(p1, p2);
+        }
+
+        [Theory, AutoData]
+        public void FreezeFirstParameterByParameterNameShouldAssignSameInstanceToSecondParameter(
+            [Freeze(By = Matching.ParameterName, TargetName = "parameter")]string p1,

```

Comment:
Why do we have to supply `TargetName` here? Why can't we just write the following?

```C#
[Theory, AutoData]
public void FreezeFirstParameterByParameterNameShouldAssignSameInstanceToSecondParameter(
    [Freeze(By = Matching.Parameter)]string parameter,
    SingleParameterType<object> p2)
{
    Assert.Equal(parameter, p2.Parameter);
}
```